graphql2/graphqlapp/alert.go

```diff
@@ -39,6 +41,36 @@ func (a *AlertLogEntry) Message(ctx context.Context, obj *alertlog.Entry) (strin
 	return e.String(), nil
 }
 
+func (a *AlertLogEntry) State(ctx context.Context, obj *alertlog.Entry) (*graphql2.AlertLogEntryState, error) {
+	e := *obj
+	meta, ok := e.Meta().(*alertlog.NotificationMetaData)
+	if !ok {
+		return nil, nil
+	}
+
+	s, err := (*App)(a).FindOneAlertLogMessageStatus(ctx, meta.MessageID)
+	if err != nil {
+		return nil, errors.Wrap(err, "find alert log state")
+	}
+
+	status := ""
+	switch s.State {
+	case notification.MessageStateFailedPerm:
+		status = "ERROR"
+	case notification.MessageStateDelivered:
+		status = "OK"
+	}
+
+	if s.Details == "" {
+		return nil, nil
+	}
```

Comment:
We can probably drop this -- we could check details & status and return nil if empty, but I don't think we should make the API depend on that.

UI can just do `state?.details` if we want to do this check. I don't think we should have a case where we have a failure but no message, but it wouldn't change the fact that it had still failed from an API standpoint

Reply:
This is a remnant from before I refactored, without this check it was returning

```
state: {
  details: "",
  status: "",
}
```
Looks like it's good to remove now :+1: